pkg/R/RDD.R

```diff
@@ -1039,6 +1039,60 @@ setMethod("keyBy",
             lapply(rdd, apply.func)
           })
 
+#' Return a new RDD that has exactly numPartitions partitions.
+#' Can increase or decrease the level of parallelism in this RDD. Internally,
+#' this uses a shuffle to redistribute data.
+#' If you are decreasing the number of partitions in this RDD, consider using
+#' coalesce, which can avoid performing a shuffle.
+#'
+#' @param rdd The RDD.
+#' @param numPartitions Number of partitions to create.
+#' @rdname repartition
+#' @seealso coalesce
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' rdd <- parallelize(sc, list(1, 2, 3, 4, 5, 6, 7), 4L)
+#' numPartitions(rdd)                   # 4
+#' numPartitions(repartition(rdd, 2L))  # 2
+#'}
+setGeneric("repartition", function(rdd, numPartitions) { standardGeneric("repartition") })
+
+#' @rdname repartition
+#' @aliases repartition,RDD
+setMethod("repartition",
+          signature(rdd = "RDD", numPartitions = "integer"),
+          function(rdd, numPartitions) {
+            jrdd <- callJMethod(getJRDD(rdd), "repartition", numPartitions)
```

Comment:
We can just use the partitionId as the seed inside `mapPartitionsWithIndex`. I think that should ensure a different random seed in each partition.